core/src/main/java/io/confluent/rest/RestCustomRequestLog.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2014 - 2022 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.rest;
+
+import java.util.Map;
+import java.util.SortedMap;
+import java.util.TreeMap;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import org.apache.kafka.common.MetricName;
+import org.apache.kafka.common.MetricNameTemplate;
+import org.apache.kafka.common.metrics.Metrics;
+import org.apache.kafka.common.metrics.Sensor;
+import org.apache.kafka.common.metrics.Sensor.RecordingLevel;
+import org.apache.kafka.common.metrics.stats.CumulativeCount;
+import org.apache.kafka.common.metrics.stats.Rate;
+import org.apache.kafka.common.metrics.stats.WindowedCount;
+import org.eclipse.jetty.server.CustomRequestLog;
+import org.eclipse.jetty.server.Request;
+import org.eclipse.jetty.server.Response;
+
+public class RestCustomRequestLog extends CustomRequestLog {
+
+  private static final long SENSOR_EXPIRY_SECONDS = TimeUnit.HOURS.toSeconds(1);
+  private static final String GROUP_NAME = "jersey-metrics";
+
+  private Sensor fourTwoNineSensor = null;
+
+  public RestCustomRequestLog(final Writer writer, final String formatString, Metrics metrics,
+      Map<String, String> metricTags, String jmxPrefix) {
+    super(writer, formatString);
+
+    if (metrics != null) {
```

Comment:
Another approach to add custom-metrics at the Jetty-layer, would be add another jetty-handler much like RequestLogHandler.java (used in [Application.java](https://github.com/confluentinc/rest-utils/blob/2e8e92703a2d386fa75bf43ccfd677c85b2ffc44/core/src/main/java/io/confluent/rest/Application.java#L376)). This would avoid _extending_ CustomRequestLog for non-logging related code.

What do you think?

PS - Since this PR is ready, it might be worth following up with another PR. OR Add a Jira tracker & a comment where in next person who _extends_ RestCustomRequestLog for non-logging related logic, can add the handler.

Reply:
That's a good call Mayank, thanks for the idea.  We shouldn't mix functions. 

Code has been updated.

Reply:
Gr8, tx! Pointed a minor nit, otherwise good to go.